selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceProjects.java

```diff
@@ -11,39 +11,148 @@
+    String SEARCH_FIELD_XPATH = "//workspace-details-projects//input[@placeholder='Search']";
+    String NOTIFICATION_CONTAINER_ID = "che-notification-container";
+    String DELETE_BUTTON_XPATH = "//che-button-primary[@che-button-title='Delete']";
+    String DELETE_BUTTON_MESSAGE_XPATH = "//che-list-header-delete-button/span";
+    String SELECT_ALL_CHECKBOX_XPATH = "//md-checkbox[@aria-label='Project list']/div";
+    String CHECKBOX_BY_PROJECT_NAME_XPATH_TEMPLATE = "//md-checkbox[@aria-label='Project %s']";
+  }
+
+  public enum BottomButton {
+    SAVE_BUTTON("save-button"),
+    APPLY_BUTTON("apply-button"),
+    CANCEL_BUTTON("cancel-button");
+
+    private final String buttonNameAttribute;
+
+    BottomButton(String buttonNameAttribute) {
+      this.buttonNameAttribute = buttonNameAttribute;
+    }
+
+    public String get() {
+      return this.buttonNameAttribute;
+    }
+  }
+
+  public void clickOnSelectAllCheckbox() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(SELECT_ALL_CHECKBOX_XPATH));
+  }
+
+  public WebElement waitDeleteButton() {
+    return seleniumWebDriverHelper.waitVisibility(By.xpath(DELETE_BUTTON_XPATH));
+  }
+
+  public void waitDeleteButtonDisappearance() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(DELETE_BUTTON_XPATH));
+  }
+
+  public void clickOnDeleteButton() {
+    waitDeleteButton().click();
+  }
+
+  public void waitDeleteButtonMessage(String expectedMessage) {
+    seleniumWebDriverHelper.waitTextContains(
+        By.xpath(DELETE_BUTTON_MESSAGE_XPATH), expectedMessage);
+  }
+
+  public void waitDeleteButtonMessageDisappearance() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(DELETE_BUTTON_MESSAGE_XPATH));
+  }
+
+  public void waitDeleteButtonDisabling() {
+    waitButtonState(By.xpath(DELETE_BUTTON_XPATH), false);
+  }
+
+  public void waitDeleteButtonEnabling() {
+    waitButtonState(By.xpath(DELETE_BUTTON_XPATH), true);
+  }
+
+  public void typeToSearchField(String text) {
+    seleniumWebDriverHelper.setValue(waitSearchField(), text);
+  }
+
+  public void waitNotification(String expectedText) {
+    seleniumWebDriverHelper.waitTextContains(By.id(NOTIFICATION_CONTAINER_ID), expectedText);
+  }
+
+  public WebElement waitBottomButton(BottomButton button) {
+    return seleniumWebDriverHelper.waitVisibility(By.name(button.get()));
+  }
+
+  public void waitBottomButtonInvisibility(BottomButton... bottomButtons) {
+    asList(bottomButtons)
+        .forEach(button -> seleniumWebDriverHelper.waitInvisibility(By.name(button.get())));
+  }
+
+  public void clickOnBottomButton(BottomButton button) {
+    waitBottomButton(button).click();
+  }
+
+  private void waitBottomButtonState(BottomButton button, boolean state) {
+    waitButtonState(By.name(button.get()), state);
+  }
+
+  private void waitButtonState(By locator, boolean state) {
+    seleniumWebDriverHelper.waitAttributeEqualsTo(
+        locator, "aria-disabled", Boolean.toString(!state));
```

Comment:
`"aria-disabled"` looks unclear here. Is it kind of `elementStateAttribute`?

Reply:
this is an attribute of web element according to the method signature

Reply:
It's not quit clear what type of value that attribute holds. It's important to know, for example, in case of the attribute name will be changed in future.